packages/eslint-plugin/src/rules/prefer-readonly.ts

```diff
@@ -105,7 +105,7 @@ export default createRule<Options, MessageIds>({
     function isDestructuringAssignment(
       node: ts.PropertyAccessExpression,
     ): boolean {
-      let current: ts.Node = node.parent;
+      let current = node.parent as ts.Node | undefined;
```

Comment:
i wonder if we should patch the TS types here?

Reply:
Yeah that would be nice. I've played around with trying to make parent optional once in a while but it's pretty much blocked on https://github.com/microsoft/TypeScript/issues/56275. I'll note that here.